File: wlr/OGUR/OGUR/Factory/ItemFactory.cs
using System;
using OGUR.Creatures;
using OGUR.GameObjects;
using OGUR.Items;

namespace OGUR.Factory
    
{
    public class ItemFactory : GameplayObject
    {
        private static Random rand = new Random();
        public static GenericItem CreateRandomPlain()
        {
            return new GenericItem(new Stats(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0),ItemSuffix.NULL,ItemPrefix.NULL, SelectRandomType());
        }
        public static GenericItem CreateRandomMagic()
        {
            return null;
            //return new IItem(new Stats(1,1,1,1,1,1,0,1,0,0,0),)            }
        }
        private static ItemType SelectRandomType()
        {
            return (ItemType)Enum.GetValues(typeof (ItemType)).GetValue(rand.Next(0, Enum.GetValues(typeof (ItemType)).Length));
        }
    }
}

File: port-workspace/aigilas-ported/src/com/aigilas/items/ItemFactory.java
package com.aigilas.items;import com.xna.wrapper.*;import java.util.*;import com.aigilas.creatures.*;import com.spx.core.*;import com.spx.entities.*;
    
    public class ItemFactory
    {
        private static final int itemGrowth = 3;
        private static int itemType;

        public static GenericItem CreateRandomPlain(Point2 location,boolean onFloor) throws Exception
        {
            if(location == null)
            {
                location = new Point2(-100,-100);
            }

            itemType = SelectRandomType();
            return
                (GenericItem)
                EntityManager.addObject(new GenericItem(new Stats(RNG.Rand.nextInt(itemGrowth), RNG.Rand.nextInt(itemGrowth), RNG.Rand.nextInt(itemGrowth), 0, 0, 0, 0, 0, 0, 0),
                                                                ItemSuffix.NULL, ItemPrefix.NULL, itemType,
                                                                location,onFloor));
        }        
        public static GenericItem CreateRandomPlain(Point2 location) throws Exception
        {
            return CreateRandomPlain(location, true);
        }        public static GenericItem CreateRandomPlain() throws Exception        {            return CreateRandomPlain(null,false);        }
        public static GenericItem CreateRandomMagic()
        {
            return null;
            //return new IItem(new Stats(1,1,1,1,1,1,0,1,0,0,0),)}
        }
        private static int SelectRandomType()
        {
            return ItemName.Values[(RNG.Next(1, ItemName.Values.length))];
        }
}